Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/LibraryBuilder.java

```diff
@@ -2935,6 +3043,60 @@ private DataType getExpressionDefResultType(ExpressionDef expressionDef) {
         throw new IllegalArgumentException(String.format("Invalid context reference from %s context to %s context.", currentExpressionContext(), expressionDef.getContext()));
     }
 
+    /**
+     * Add an identifier to the deque to indicate that we are considering it for consideration for identifier hiding and
+     * adding a compiler warning if this is the case.
+     * <p/>
+     * For example, if an alias within an expression body has the same name as a parameter, execution would have
+     * added the parameter identifier and the next execution would consider an alias with the same name, thus resulting
+     * in a warning.
+     * <p/>
+     * Exact case matching as well as case-insensitive matching are considered.  If known, the type of the structure
+     * in question will be considered in crafting the warning message, as per the {@link Element} parameter.
+     *
+     * @param identifier The identifier belonging to the parameter, expression, function, alias, etc, to be evaluated.
+     * @param onlyOnce Special case to deal with overloaded functions, which are out scope for hiding.
+     * @param element The consturct element, for {@link ExpressionRef}.
+     * @param nullableExpression Use strictly to comply with the signature for {@link #reportWarning(String, Expression)}.
```

Comment:
issue: I think we can reuse the existing Defs (which are Elements) rather than creating Refs (which are Expressions). The means we don't need the final parameter for this function. We can simply relax the signature of reportWarning from `Expression` -> `Element`. The trackbacks are available there.

Reply:
After changing the signature to accept Elements, I got rid of all the code I introduced to deal with Expressions, including the "get"/"resolve" methods you commented on above.